packages/rocketchat-livechat/server/models/LivechatDepartmentAgents.js

```diff
@@ -70,6 +70,34 @@ class LivechatDepartmentAgents extends RocketChat.models._Base {
 			return null;
 		}
 	}
+
+	getForDepartment(departmentId) {
+		var agents = this.findByDepartmentId(departmentId).fetch();
+
+		if (agents.length === 0) {
+			return;
+		}
+
+		var onlineUsers = RocketChat.models.Users.findOnlineUserFromList(_.pluck(agents, 'username'));
+
+		var onlineUsernames = _.pluck(onlineUsers.fetch(), 'username');
+
+		var query = {
+			departmentId: departmentId,
+			username: {
+				$in: onlineUsernames
+			}
+		};
+
+		var agents = this.find(query);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["agents" is already defined](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2726640534/issues/source?bid=3199634&fileBranchId=3348698#l92)